prophy/composite.py

```diff
@@ -584,3 +594,12 @@ def __init__(cls, name, bases, attrs):
             extend_union_descriptor(cls, descriptor)
             add_union_properties(cls, descriptor)
         super(union_generator, cls).__init__(name, bases, attrs)
+
+
+
+class Descriptor(object):
+
+    def __init__(self,name, type_, kind):
+            self.name = name
+            self.kind = kind
+            self.type_ = type_
```

Comment:
Consider changing self.type_ name to self.type.
PEP8 argues that names colliding with built-ins and keywords should be trailed with underscore, but in case of class member you don't get shadowing as you have to name enclosing object first:
```
type_  # this - as function argument - will shadow type builtin. Bad idea
self.type  # this doesn't shadow type built-in
field.type  # neither does this
```